components/DetectAnimatedGif.coffee

```diff
@@ -22,7 +22,27 @@ exports.getComponent = ->
     async: true
     forwardGroups: true
   , (buffer, groups, out, callback) ->
-    out.send animated buffer
-    do callback
+    if Buffer.isBuffer buffer
+      out.send animated buffer
+      do callback
+      return
+    else if typeof buffer is 'string'
+      isAnimated = false
+      fs.createReadStream buffer
+        .pipe animated()
+        .once 'animated', ->
+          isAnimated = true
+          out.send true
+          do callback
+          return
+        .on 'finish', ->
+          unless isAnimated
```

Comment:
Isn't reading of the GIF (and all its frames if animated) syncronous process? I don't see anywhere where we abort this in case of `isAnimated: true`?
Assuming this is true, there is no 'perf impact'.